### PR TITLE
Clear auth cache on user management operations

### PR DESCRIPTION
**Pull request by henriknyman**
The user management operations in the internal Shiro realm needs to clear
the relevant cached information.

### CODE REVIEW

**Comment by craigtaverner** [comment]
Is it possible to write a test that would fail without this change?

**Comment by henriknyman** [comment]
AuthProceduresTest userCreation2, userCreation3, userCreation4 and roleManagement2 fails without it

File: enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java [hunk 1]
```diff
@@ -217,6 +218,7 @@ void addUserToRole( String username, String roleName ) throws IOException
                 }
             }
         }
+        clearCachedAuthorizationInfoForUser( username );
     }
 
     void removeUserFromRole( String username, String roleName ) throws IOException
```
Comment (OliviaYtterbrink) [inline enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java:237]: `removeUserFromAllRoles` should also clear authorization info.

Comment (henriknyman) [inline enterprise/security/src/main/java/org/neo4j/server/security/enterprise/auth/FileUserRealm.java:237]: The thinking was that removeUserFromAllRoles is private so as long as all exposed methods that use it clears the cache that should be enough

**Pull request merged**